scripts/build/get_source/termux_unpack_src_archive.sh

```diff
@@ -13,6 +13,8 @@ termux_extract_src_archive() {
 			rm -Rf $folder
 			unzip -q "$file"
 			mv $folder "$TERMUX_PKG_SRCDIR"
+		elif [ "${file##*.}" = patch ] || [ "${file##*.}" = diff ]; then
+			continue
```

Comment:
What if, instead of using this type of condition (test for `.patch` and `.diff` filenames),
a different form of condition were used like this, which uses a different logic to make sure `tar` is not used when not appropriate?

```diff
--- a/scripts/build/get_source/termux_unpack_src_archive.sh
+++ b/scripts/build/get_source/termux_unpack_src_archive.sh
@@ -13,7 +13,7 @@ termux_extract_src_archive() {
 			rm -Rf $folder
 			unzip -q "$file"
 			mv $folder "$TERMUX_PKG_SRCDIR"
-		else
+		elif file "$file" | grep -q tar; then
 			test "$i" -gt 0 && STRIP=0
 			mkdir -p "$TERMUX_PKG_SRCDIR"
 			tar xf "$file" -C "$TERMUX_PKG_SRCDIR" --strip-components=$STRIP
```

Reply:
If you're going to determine it based of the filename please use `[[` bash test `]]`, quote the right hand sides and inline the `||`.
```suggestion
		elif [[ "${file##*.}" = 'patch' || "${file##*.}" = 'diff' ]]; then
			continue
```

I agree with Robert though that we should probably be using a more rigorous method to determine archive type.
`file --brief --mime-type` may be a good option for this but I haven't tested it thoroughly.

Reply:
Initially I tested `file`, but it prints a different result for every different subtype of `.tar.xxx` archive, for example for `.tar.gz` files it prints `gzip`, for `.tar.bz2` files it prints `z-bzip2`, for `.tar.xz` files it prints `x-xz`, and so on, so I thought that a way to do it without creating a bunch more conditions to try to test for every possible type of `.tar.xxx` archive, might be to use `tar` itself to auto detect whether the file type is valid using the same code internal to the `tar` command that is used to auto detect and call the correct extraction backend later at the `tar xf` line.

Reply:
The source code of the `makepkg` command of Arch Linux does actually use `file` and then test for every one of those individually, pretty much like you suggest, but then also falls back to calling `bsdtar -tf`  in a very similar way to my suggestion, so it's like they combined both concepts. I am not sure why it's important for them to do both. Maybe it improves logging for them, so that they can always see an explicit log of the exact file type and extraction program used.

https://gitlab.archlinux.org/pacman/pacman/-/blob/master/scripts/libmakepkg/source/file.sh.in#L99-L142

Reply:
Since it was also mentioned by biswa96 through the mention of `APKBUILD`  as well as `PKGBUILD` :
Alpine Linux's analogous code doesn't use `file` or `tar -tf` at all, it just relies entirely on the actual filename, and explicitly checks for every possible file extension. 

https://gitlab.alpinelinux.org/alpine/abuild/-/blob/857d4c58963d14d637ee4e35b9b335ad14cb424e/abuild.in#L506-L553

This means there are kind of a lot of ways to write this and probably there isn't a single "objectively correct" solution, it's whatever the distro maintainer feels looks best.

Reply:
I tracked down the original reason given that Arch Linux began using the `file` command instead of the file name extension to determine which extraction command to use.

https://web.archive.org/web/20070220072658/http://bbs.archlinux.org/viewtopic.php?t=29031

It happened in 2007 when someone wanted to package "a program distributed as a `.tar.gz` source archive, except that it's really a `.tar` archive". The exact tarball name or project responsible for the incorrect filename involved  was never given, so I have no idea if the issue could still be carefully reproduced today in Alpine Linux, but I would tentatively guess that most likely, the issue was eventually also reported to whatever project released that tarball, and any modern continuation of that project most likely has been releasing better-formed tarballs for years, which is probably why Alpine Linux doesn't need to do that.

Reply:
Please feel free to create another pull request with the appropriate logic. This is the shortest change I came up with to get PKGBUILD like syntax.